Splitio.Redis/Services/Cache/Classes/RedisCacheBase.cs

```diff
@@ -10,33 +14,45 @@ public abstract class RedisCacheBase
         protected string MachineIp;
         protected string MachineName;
 
-        public RedisCacheBase(string userPrefix = null)
+        protected RedisCacheBase(RedisConfig redisConfig = null, bool clusterMode = false)
         {
-            UserPrefix = userPrefix;
+            UserPrefix = redisConfig.RedisUserPrefix;
             RedisKeyPrefix = "SPLITIO.";
-
-            if (!string.IsNullOrEmpty(userPrefix))
-            {
-                RedisKeyPrefix = userPrefix + "." + RedisKeyPrefix;
-            }
+            UpdateRedisKeyPrefix(UserPrefix, redisConfig, clusterMode);
         }
 
         public RedisCacheBase(string machineIP,
             string sdkVersion,
             string machineName,
-            string userPrefix = null)
+            RedisConfig redisConfig = null, bool clusterMode = false)
         {
-            UserPrefix = userPrefix;
+            UserPrefix = redisConfig.RedisUserPrefix;
             MachineIp = machineIP;
             SdkVersion = sdkVersion;
             MachineName = machineName;
 
             RedisKeyPrefix = RedisKeyPrefixFormat
                 .Replace("{sdk-language-version}", sdkVersion)
                 .Replace("{instance-id}", machineIP);
+            UpdateRedisKeyPrefix(UserPrefix, redisConfig, clusterMode);
+        }
 
+        private void UpdateRedisKeyPrefix(string userPrefix, RedisConfig redisConfig, bool clusterMode)
+        {
             if (!string.IsNullOrEmpty(userPrefix))
             {
+                if (clusterMode)
+                {
+                    if (redisConfig.ClusterNodes != null && !string.IsNullOrEmpty(redisConfig.ClusterNodes.KeyHashTag))
+                    {
+                        userPrefix = redisConfig.ClusterNodes.KeyHashTag + userPrefix;
+                    }
+                    else
+                    {
+                        userPrefix = KeyHashTag + userPrefix;
+                    }
+                    UserPrefix = userPrefix;
+                }
                 RedisKeyPrefix = userPrefix + "." + RedisKeyPrefix;
             }
```

Comment:
This method is wrong, because you can run in cluster mode without userPrefix, so the first if is not correct.
 Also you can split this method in 2
  - one to set redis key prefix
  - one to set hashtag if is cluster mode

Reply:
Will change it to the following:
if (!string.IsNullOrEmpty(userPrefix))
{
     userPrefix = "";
}
if (clusterMode) 
{ 
...
}